sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/trees/TreeNode.scala

```diff
@@ -544,6 +545,8 @@ abstract class TreeNode[BaseType <: TreeNode[BaseType]] extends Product {
     case None => Nil
     case Some(null) => Nil
     case Some(any) => any :: Nil
+    case map: CaseInsensitiveStringMap =>
+      truncatedString(map.entrySet().toArray(), "[", ", ", "]", maxFields) :: Nil
```

Comment:
are you sure this returns meaningful string?

Reply:
and we should at least display the original map (keys are not lowercased)

Reply:
we can use `asCaseSensitiveMap` call.

Reply:
I tried the following
```
scala> Map("a" -> 1).asJava.entrySet.toArray.head
res1: Object = scala.collection.convert.Wrappers$MapWrapper$$anon$1$$anon$5$$anon$6@60
```

I think it's safer to generate the string `key=value` manually.

Reply:
I tried this:
```
scala> val options = Map(
     | "Key1" -> "Value1",
     | "key2" -> "vaue2"
     | )
options: scala.collection.immutable.Map[String,String] = Map(Key1 -> Value1, key2 -> vaue2)

scala> val map = new CaseInsensitiveStringMap(options.asJava)
map: org.apache.spark.sql.util.CaseInsensitiveStringMap = org.apache.spark.sql.util.CaseInsensitiveStringMap@9f174175

scala> map.asCaseSensitiveMap().entrySet().toArray()
res3: Array[Object] = Array(Key1=Value1, key2=vaue2)
```
I used `case map: CaseInsensitiveStringMap` for pattern matching, I guess we are probably OK?

Reply:
Can we add a test case in `ExplainSuite`?

Reply:
Sure. Will do. Have a good night!

Reply:
Seems we can't have a ```UnresolvedRelation``` with a non empty ```CaseInsensitiveStringMap``` in explain? The only way to create a ```UnresolvedRelation``` with a non empty ```CaseInsensitiveStringMap``` is using ```DataFrameReader.table```, but this is resolved to a ```V2Relation```.

Reply:
`df.explain(true)` will explain the parsed plan, which contains `UnresolvedRelation`

Reply:
added a test. Thank you!